cms/djangoapps/contentstore/tests/test_course_settings.py

```diff
@@ -119,6 +122,34 @@ def test_mobile_field_available(self):
         self.assertEqual(settings_fields["display_name"], "Mobile Course Available")
         self.assertEqual(settings_fields["deprecated"], True)
 
+    @ddt.data(
+        (False, False),
+        (True, False),
+        (True, True)
+    )
+    @ddt.unpack
+    def test_discussion_fields_available(self, is_pages_and_resources_enabled, is_legacy_discussion_setting_enabled):
```

Comment:
let's have a 3rd argument here which would be `fields_visible` flag. Having that flag in the ddt would help us remove the if condition in the test body. 


```
@ddt.data(
  (False, False, True),
  (True, False, False),
  (False, True, True),
  (True, True, True)
)
@ddt.unpack
def test_discussion_fields_available(self, is_mfe_enabled, is_discussion_setting_overriden, fields_visible):
```

After this, the body of the test can be updated to
```
self.assertEqual('allow_anonymous' in response, fields_visible)
```